attest/ake/src/lib.rs

```diff
@@ -7,6 +7,7 @@
 //! (client) and responders (server).
 
 #![allow(clippy::type_complexity)]
+#![allow(clippy::result_large_err)]
```

Comment:
:thought_balloon: TL;DR this is fine for this PR of updating rust, just curious if we want to follow up later.

It looks like without the suppression clippy outputs:
```console
error: the `Err`-variant returned from this function is very large
  --> attest/ake/src/responder.rs:39:10
   |
39 |     ) -> Result<(Ready<Cipher>, AuthResponseOutput), Error>
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
  ::: attest/ake/src/error.rs:28:5
   |
28 |     ReportVerification(VerifierError),
   |     --------------------------------- the largest variant contains at least 696 bytes
```

we may not be too worried that we're always returning a 700byte structure, but I wonder if we should consider creating a follow on issue to consider addressing the error size.

There is also a configuration value if we agree that the default size is too restrictive
> Configuration
>
> This lint has the following configuration variables:
>
>    large-error-threshold: u64(defaults to 128): The maximum size of the Err-variant in a Result returned from a function

Reply:
@jcape and I discussed this, and we think that this is not worth pursuing. Boxing these errors comes at a significant code readability cost- a lot of the error consumption code destructures these errors. If you box them, the inner field (i.e. what used to be just the error) is now an opaque (box) type, which prevents destructuring and makes you do some ugly box dereferences. 

There's also no indication that the size of these errors has any impact on the performance of our code. So given these two things, we decided to not implement it right now. If we find that it becomes a problem in the future, then we can reconsider.